lighthouse-core/audits/audit.js

```diff
@@ -65,9 +64,9 @@ class Audit {
   }
 
   /**
-   * @param {!Audit} audit
+   * @param {typeof Audit} audit
```

Comment:
what's this mean btw?

Reply:
`Audit` is the name of the class in js-land, but in type-land it's the name of the type of *instances* of `Audit`.

To refer to the type of the class `Audit` itself (since we use all static methods) you have to use `typeof`, which is kind of unfortunately overloaded in TS but gets the job done and isn't *too* obscure

Reply:
huh! thanks for explaining.
wish it was `instanceof` instead.

Reply:
Fortunately you almost always the normal thing. It's just here we're not just using all static methods, we want the static methods on subclasses of `Audit`, so we have to pass the subclass in to `Audit`'s static methods... 🤪